src/databricks/sql/auth/oauth.py

```diff
@@ -0,0 +1,286 @@
+# Databricks CLI
+# Copyright 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import base64
+import hashlib
+import json
+import os
+import webbrowser
+
+from datetime import datetime, timedelta, tzinfo
+
+import click
+
+import jwt
+from jwt import PyJWTError
+
+import oauthlib.oauth2
+from oauthlib.oauth2.rfc6749.errors import OAuth2Error
+
+import requests
+from requests.exceptions import RequestException
+
+from databricks_cli.utils import error_and_quit
```

Comment:
you're not allowed to use it outside of cli package. it calls for `sys.exit`, which is end of the process. use proper exception re-wrapping.

https://github.com/databricks/databricks-cli/blob/main/databricks_cli/utils.py#L94-L100